scarb-metadata/src/command/metadata_command.rs

```diff
@@ -184,6 +184,7 @@ fn parse_stream(data: &[u8]) -> Result<Metadata, MetadataCommandError> {
 }
 
 #[cfg(test)]
+#[allow(clippy::redundant_pattern_matching)]
```

Comment:
Is this caused by usage of `matches!` macro? Can we put this right before the line which is triggering this issue?